README.pod
package HTML::Escape;
use strict;
use warnings;
use 5.008008;
our $VERSION = '1.06';
use parent qw/Exporter/;

our $TESTING_PERL_ONLY;
$TESTING_PERL_ONLY = $ENV{HTML_ESCAPE_PUREPERL} ? 1 : 0
        unless defined $TESTING_PERL_ONLY;

unless($TESTING_PERL_ONLY){
    local $@;
    $TESTING_PERL_ONLY = !eval{
        require XSLoader;
        XSLoader::load(__PACKAGE__, $VERSION);
    };
}
if ($TESTING_PERL_ONLY) {
    ## no critic.
    require 'HTML/Escape/PurePerl.pm' # not to create the namespace
}

our @EXPORT = qw/escape_html/;

1;
__END__

=encoding utf8

=head1 NAME

HTML::Escape - Extremely fast HTML escape

=head1 SYNOPSIS

    use HTML::Escape qw/escape_html/;

    escape_html("<^o^>");

=head1 DESCRIPTION

This module escapes HTML's special chars. It's same as PHP's htmlspecialchars.

This module uses XS for better performance.
And also provides pure perl version.

=head1 BENCHMARK

                     Rate HTML::Entities   HTML::Escape
    HTML::Entities 14.0/s             --           -91%
    HTML::Escape    150/s           975%             --

=head1 AUTHOR

Goro Fuji

Tokuhiro Matsuno E<lt>tokuhirom AAJKLFJEF@ GMAIL COME<gt>

=head1 SEE ALSO

L<Text::Xslate>, L<HTML::Entities>

=head1 LICENSE

Copyright (C) Tokuhiro Matsuno

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut


Changes
Revision history for Perl extension HTML::Escape

{{$NEXT}}

1.11 2021-09-27T04:11:20Z

    commit 239b88f865305b59f7d193f0431fcd5c03df3dd3
    Author: Harald Jörg <haj@posteo.de>
    Date:   Sun Jan 13 11:56:25 2019 +0100

        Add the list of escaped characters to the docs and clarify usage in the synopsis


1.10 2016-03-15T03:16:39Z

    - Performance improvement.
      Aaron Crane <arc@cpan.org>

1.09 2014-04-14T07:34:14Z

    - Escape '{' && '}' by default.
      https://github.com/angular/angular.js/issues/5601
      (tokuhirom)

1.08 2013-06-08T16:02:51Z

    - Escape backquote character for IE
      (Requested by John Siracusa)
      https://rt.cpan.org/Public/Bug/Display.html?id=84971

1.07 2013-06-08T04:59:08Z

    - use Minilla(no feature changes)

1.06

    - use M::B::Pluggable(No feature changes)

1.05

    - Fixed ppport.h generation logic.
      (tokuhirom)

1.04

    - fixed typo in dependency list.
      (Reported by syohex++)

1.03

	- depend to perl 5.8.5, gfx was tested on it.
      (gfx)

1.02

    [BUG FIX]
    - works on ithread enabled perl.

1.01

    - use Text::Xslate style pure perl module loading

1.00
    
    [INCOMPATIBLE CHANGE]
    - s/html_escape/escape_html/g

0.01    Sun Sep 30 15:09:52 2012
    - original version


Build.PL
# =========================================================================
# THIS FILE IS AUTOMATICALLY GENERATED BY MINILLA.
# DO NOT EDIT DIRECTLY.
# =========================================================================

use 5.008_001;

use strict;
use warnings;
use utf8;

use Module::Build;
use File::Basename;
use File::Spec;
use CPAN::Meta;
use CPAN::Meta::Prereqs;

my %args = (
    license              => 'perl',
    dynamic_config       => 0,

    configure_requires => {
        'Module::Build' => 0.4005,
    },

    name            => 'HTML-Escape',
    module_name     => 'HTML::Escape',
    allow_pureperl => 1,

    script_files => [glob('script/*'), glob('bin/*')],
    c_source     => [qw()],
    PL_files => {},

    test_files           => ((-d '.git' || $ENV{RELEASE_TESTING}) && -d 'xt') ? 't/ xt/' : 't/',
    recursive_test_files => 1,

    
);
if (-d 'share') {
    $args{share_dir} = 'share';
}

my $builder = Module::Build->subclass(
    class => 'MyBuilder',
    code => q{
        sub ACTION_distmeta {
            die "Do not run distmeta. Install Minilla and `minil install` instead.\n";
        }
        sub ACTION_installdeps {
            die "Do not run installdeps. Run `cpanm --installdeps .` instead.\n";
        }
    }
)->new(%args);
$builder->create_build_script();

my $mbmeta = CPAN::Meta->load_file('MYMETA.json');
my $meta = CPAN::Meta->load_file('META.json');
my $prereqs_hash = CPAN::Meta::Prereqs->new(
    $meta->prereqs
)->with_merged_prereqs(
    CPAN::Meta::Prereqs->new($mbmeta->prereqs)
)->as_string_hash;
my $mymeta = CPAN::Meta->new(
    {
        %{$meta->as_struct},
        prereqs => $prereqs_hash
    }
);
print "Merging cpanfile prereqs to MYMETA.yml\n";
$mymeta->save('MYMETA.yml', { version => 1.4 });
print "Merging cpanfile prereqs to MYMETA.json\n";
$mymeta->save('MYMETA.json', { version => 2 });


lib/HTML/Escape.pm
package HTML::Escape;
use strict;
use warnings;
use 5.008005;
our $VERSION = '1.10';
use parent qw/Exporter/;

my $use_xs = 0;
if(!exists $INC{'HTML/Escape/PurePerl.pm'}) {
    my $pp = $ENV{PERL_ONLY};
    if (!$pp) {
        eval {
            require XSLoader;
            XSLoader::load(__PACKAGE__, $VERSION);
            $use_xs = 1;
        };
    }
    if (!__PACKAGE__->can('escape_html')) {
        ## no critic.
        require 'HTML/Escape/PurePerl.pm' # not to create the namespace
    }
}
sub USE_XS () { $use_xs }

our @EXPORT = qw/escape_html/;

1;
__END__

=encoding utf8

=head1 NAME

HTML::Escape - Extremely fast HTML escaping

=head1 SYNOPSIS

    use HTML::Escape qw/escape_html/;

    my $escaped = escape_html("<^o^>");

=head1 DESCRIPTION

This modules provides a function which escapes HTML's special characters. It
performs a similar function to PHP's htmlspecialchars.  It escapes the
following characters:

    " & ' < > ` { }

This module uses XS for better performance, but it also provides a pure perl
version.

=head1 FAQ

=over 4

=item Is there also an unescape_html?

No. Unescaping HTML requires a lot of code, and we don't want to do it.
Please use L<HTML::Entities> for it.

=back

=head1 BENCHMARK

                     Rate HTML::Entities   HTML::Escape
    HTML::Entities 14.0/s             --           -91%
    HTML::Escape    150/s           975%             --

=head1 AUTHOR

Goro Fuji

Tokuhiro Matsuno E<lt>tokuhirom AAJKLFJEF@ GMAIL COME<gt>

=head1 SEE ALSO

L<Text::Xslate>, L<HTML::Entities>

=head1 LICENSE

Copyright (C) Tokuhiro Matsuno

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut
